### ISSUE TITLE
OBSERVATION with an "empty" observation causes an endless loop in validator

### ISSUE THREAD

**Issue by chevalleyc** [openEHR/archie#323]
The attached composition when validated ends up in an endless loop when constructing the error (?) message.

The composition is validated using

`new RMObjectValidator(ArchieRMInfoLookup.getInstance()).validate(composition)`

and the following snippet results in the loop:

```
   private List<RMObjectValidationMessage> isObservationEmpty(CAttribute attribute, String rmAttributeName, Object attributeValue, String pathSoFar, CObject cobject) {
        List<RMObjectValidationMessage> result = new ArrayList();
        CObject parent = attribute.getParent();
        boolean parentIsEvent = parent != null && parent.getRmTypeName().contains("EVENT");
        boolean attributeIsData = rmAttributeName.equals("data");
        boolean attributeIsEmpty = attributeValue == null;
        boolean attributeShouldNotBeEmpty = attribute.getExistence() != null && !attribute.getExistence().has(0);
        if (parentIsEvent && attributeIsData && attributeIsEmpty && attributeShouldNotBeEmpty) {
--->            String message = "Observation " + RMObjectValidationUtil.getParentObservationTerm(attribute) + " contains no results";
            result.add(new RMObjectValidationMessage(cobject == null ? null : cobject.getParent().getParent(), pathSoFar, message, RMObjectValidationMessageType.EMPTY_OBSERVATION));
        }

        return result;
    }
```
Composition:

[empty-observation-composition.json.txt](https://github.com/openEHR/archie/files/6653936/empty-observation-composition.json.txt)

**Comment by pieterbos** [comment]
Thanks for the issue report!

I added a test case for this in the branch `empty_observation_endless_loop`, this this commit: https://github.com/openEHR/archie/commit/4f01244aeec2b97c8768363df51ada58a576b0d8

However, that test passes without problems.

The method you mention, getParentObservationTerm, has been updated in https://github.com/openEHR/archie/commit/56acbe8872a89bf645fb5bfe916ad4f586269109 . That might have fixed it?

So if you update to archie version 1.0.0 or perhaps version  0.19.0, is the problem gone?
